### PR TITLE
ddl: Revoke the session when the DDL will close

### PR DESCRIPTION
**Pull request by zimulala**
`setManagerSessionTTL` is used for testing. Setting it to a small value can reduce the waiting time for the next round of campaign leaders when the leader DDL is closed.

### CODE REVIEW

File: ddl/owner_manager.go [hunk 1]
```diff
@@ -112,6 +114,17 @@ func (m *ownerManager) SetBgOwner(isOwner bool) {
 // ManagerSessionTTL is the etcd session's TTL in seconds. It's exported for testing.
 var ManagerSessionTTL = 60
 
+// setManagerSessionTTL sets the ManagerSessionTTL value, it's used for testing.
+func setManagerSessionTTL() error {
```
Comment (shenli) [inline ddl/owner_manager.go]: Should we do this in init() for only once?

File: ddl/owner_manager.go [hunk 2]
```diff
@@ -162,8 +180,13 @@ func (m *ownerManager) campaignLoop(ctx goctx.Context, etcdSession *concurrency.
 				return
 			}
 		case <-ctx.Done():
-			log.Infof("[ddl] break %s campaign loop", key)
-			return
+			// Revoke the session lease.
+			// If revoke takes longer than the ttl, lease is expired anyway.
+			ctx, cancel := goctx.WithTimeout(goctx.Background(),
+				time.Duration(ManagerSessionTTL)*time.Second)
+			_, err = m.etcdCli.Revoke(ctx, etcdSession.Lease())
```
Comment (shenli) [inline ddl/owner_manager.go]: Why we need to do this?

File: ddl/ddl.go [hunk 3]
```diff
@@ -334,11 +334,11 @@ func (d *ddl) close() {
 	}
 
 	close(d.quitCh)
+	d.ownerManager.Cancel()
```
Comment (coocood) [inline ddl/ddl.go]: What's the difference?

File: ddl/owner_manager.go (hunk 2 above)
Comment (zimulala) [inline ddl/owner_manager.go]: Revoke the session directly, don't wait for it to expire.

File: ddl/ddl.go (hunk 3 above)
Comment (zimulala) [inline ddl/ddl.go]: Because we did not really wait for `ownerManager` to quit, so I update this.

File: ddl/owner_manager.go (hunk 1 above)
Comment (zimulala) [inline ddl/owner_manager.go]: This function has an error message that needs to be returned, so the current approach might be better.

Comment (shenli) [inline ddl/owner_manager.go]: I mean set ManagerSessionTTL according to environment variable. If Atoi returns error, we could keep ManagerSessionTTL unchanged and print a warning log.

File: ddl/owner_manager.go (hunk 2 above)
Comment (shenli) [inline ddl/owner_manager.go]: ok

Comment (coocood) [inline ddl/owner_manager.go]: Can `Revoke` be used by `schemaSyncer` instead of `removeSelfVersionPath`?

Comment (zimulala) [inline ddl/owner_manager.go]: I will update it in the next PR.

**Comment by zimulala** [comment]
PTAL @shenli @coocood

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**